### PR TITLE
Space Travelers: Mission Actions

### PR DESCRIPTION
**Pull request by dselasea**
In this pull request I
- added action for mission
- added action in the mission component
- exported actions from missionSlice

### CODE REVIEW

File: src/redux/missions/missionsSlice.js [hunk 1]
```diff
@@ -20,10 +20,17 @@ const initialState = {
...
   reducers: {
-
+    joinMission: (state, action) => {
+      const id = action.payload;
+      const missions = state.missions.map((mission) => {
+        if (mission.id !== id) return mission;
+        return { ...mission, reserved: true };
+      });
+      return { ...state, missions };
+    },
```
Comment (preciousbetine) [inline src/redux/missions/missionsSlice.js:33]: In the `joinMission` reducer, the id for a mission is stored in the `mission_id` field.
You should consider changing line 29 to:

```
if (mission.mission_id !== id) return mission;
```

**Review by preciousuwhubetine — CHANGES_REQUESTED** [review]
## Hi, @dselasea 

Good job so far!
You've implemented most of the requirements for this card 🤝 . But there are some issues that need to be fixed before moving on.



Suggested changes

-  _Check the comments under the review._ ⬇

Cheers and Happy coding!👏👏👏

File: src/redux/missions/missionsSlice.js (hunk 1 above)
Comment (preciousuwhubetine) [inline src/redux/missions/missionsSlice.js]: In the `joinMission` reducer, the id for a mission is stored in the `mission_id` field.
You should consider changing line 29 to:

```
if (mission.mission_id !== id) return mission;
```

**Review by preciousuwhubetine — APPROVED** [review]
## STATUS: APPROVED :clap: 

approved-accepted

This branch can now be merged.

Highlights:

- Mission joining feature is working :heavy_check_mark:
- No linter errors :heavy_check_mark: 

Cheers and Happy coding!👏👏👏

**Pull request merged**